src/System.Text.Utf8String/System/Text/Utf8String.cs

```diff
@@ -125,14 +120,18 @@ public CodePointEnumerable CodePoints
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
             get
             {
-                // there is no need to check the boundaries -> Span is going to do this on it's own
-                return (byte)_buffer[i];
+                // Check bounds
+                if( i < 0 || i > _buffer.Length)
```

Comment:
Shouldn't we change the bounds check to i >= _buffer.Length?

Reply:
Wouldn't that lead to the index being outside of the bounds of the array? (i.e. the buffer runs from 0 to Length - 1)

Reply:
> Wouldn't that lead to the index being outside of the bounds of the array? (i.e. the buffer runs from 0 to Length - 1)

Currently, the check accounts for buffer from 0 to Length.

I might be mistaken, but if `i = _buffer.Length` and you don't throw (during bounds check), we have the following:

`return _buffer[_buffer.Length];` <= isn't this out of range?

Reply:
It is! Sorry and thanks.

Reply:
@A-And, just adding comment so that this change suggestion is not lost in the comments.